### PR TITLE
Add test to verify notifications from other sources are ignored

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm [hunk 1]
```diff
@@ -178,6 +178,44 @@ - (nonnull NSString*)tokenDidChangeNotificationName {
...
 }
 
+// Regression test for https://github.com/firebase/firebase-ios-sdk/issues/8895
+TEST(FirebaseAppCheckCredentialsProviderTest,
+     ListenForTokenChangesIgnoresUnrelatedNotifcations) {
+  auto token_promise = std::make_shared<std::promise<std::string>>();
+
+  FIRApp* app = testutil::AppForUnitTesting();
+  FSTAppCheckFake* app_check = [[FSTAppCheckFake alloc] initWithToken:@""];
+  FirebaseAppCheckCredentialsProvider credentials_provider(app, app_check);
```
Comment (maksymmalyhin) [inline Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:188]: Can we add a variant of this test and `GetToken` test with `app_check == nil`?

**Review by maksymmalyhin — APPROVED** [review]
👍  LGTM, a couple questions.

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm (hunk 1 above)
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:188]: There is already a test for this: GetTokenNoProvider.

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm [hunk 2]
```diff
@@ -178,6 +178,44 @@ - (nonnull NSString*)tokenDidChangeNotificationName {
...
+
+  credentials_provider.SetCredentialChangeListener(
+      [token_promise](const std::string& result) {
+        if (result != "") {
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:192]: The token listener fires once during the initialization with an empty token.

File: Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm [hunk 3]
```diff
@@ -178,6 +178,44 @@ - (nonnull NSString*)tokenDidChangeNotificationName {
...
+      postNotificationName:[app_check tokenDidChangeNotificationName]
+                    object:app_check
+                  userInfo:@{
+                    [app_check notificationTokenKey] : @"token2",
+                    [app_check notificationAppNameKey] : [app name],
+                  }];
+
+  auto kTimeout = std::chrono::seconds(5);
+  auto token_future = token_promise->get_future();
+  ASSERT_EQ(std::future_status::ready, token_future.wait_for(kTimeout));
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:216]: The test would fail in line 193, as the comparisons of the token value would fail. The promise does only resolve once, but it is merely used to ensure that the callback fired. 
I moved the assert of the token value out of the callback - maybe that makes it a bit cleaner.

[the remark this answers is not in the document]
Comment (maksymmalyhin) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:216]: Hm... It looks like if `@"unrelated"` notification passes through then the promise value will be still overwritten to `"token2"` as it is called later, so the test won't actually fail. Am I missing anything?

[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:216]: Promises can only be resolved once.

[the remark this answers is not in the document]
Comment (maksymmalyhin) [reply Firestore/core/test/unit/credentials/firebase_app_check_credentials_provider_test.mm:216]: Ah, sorry, I misread the comment above!
